### PR TITLE
Use system trust store instead of cacert.pem

### PR DESCRIPTION
**Pull request by bdewater**
Having a bot [update the cacert.pem file](https://github.com/Shopify/httpclient/commit/1fc910e2fecfeb7d75d3c4b889fc2037e645b292) is addressing the symptom, not the root cause IMO.

Bundling certificate files in a Ruby gem to work around buggy/old OpenSSL installations was a ~decade old solution. A lot has changed on the HTTPS front in the past five to ten years: 
- SSL certificates became cheaper and more prevalent to the point where it's now free and near-ubiquitous. See stats from [Google](https://transparencyreport.google.com/https/overview?hl=en) and [Let's Encrypt](https://letsencrypt.org/stats/).
- The Heartbleed vulnerability got the OpenSSL project the investments it badly needed. The Linux Foundation started the [Core Infrastructure Initiative](https://www.coreinfrastructure.org/) in response to it. Development picked up and many things were improved (see [this 2020 paper](https://arxiv.org/abs/2005.14242)).
- Bad certificate authorities faced the consequences of not handling the trust placed in them properly, e.g. [DigiNotar](https://en.wikipedia.org/wiki/DigiNotar) and [Symantec](https://arstechnica.com/information-technology/2017/03/google-takes-symantec-to-the-woodshed-for-mis-issuing-30000-https-certs/).

In short: HTTPS nowadays 'just works', operating systems update the trust store regularly, bundling certificates isn't needed anymore and can even cause issues if not updated for a long time, as evidenced by this fork existence. Using the system trust store should cause less surprises (parity with using Ruby Net::HTTP directly) and definitely is less churn of bumping a transitive dependency monthly that's not used by most apps directly.

### CODE REVIEW

**Comment by bdewater** [comment]
I missed this [excellent blog post](https://shopify.engineering/that-old-certificate-expired-and-started-an-outage-this-is-what-happened-next) by @XaF :sweat_smile:
> The choice of overriding cacert.pem with a more recent one is a temporary fix. However, it was the one, following a solve-fast approach, we knew would work automatically for all our deployments without making any other changes. To support this fix and make sure a similar issue does not happen soon, we put in place systems to keep track of changes in the root certificates, and automatically update them in our fork when needed. A better long-term approach could be to use the system root certificates store for instance, which can commence after a review of our system root certificate stores, across all of our runtime environments.
